keras/src/saving/saving_lib.py

```diff
@@ -8,6 +8,7 @@
 import warnings
 import zipfile
 
+import huggingface_hub
```

Comment:
I think we should probably conditional import this, with an error and install instructions if this is not available.

https://github.com/keras-team/keras-nlp/blob/9bf85235f84dcc86238d785775f393a4abb9d2dc/keras_nlp/src/utils/preset_utils.py#L171-L175

If not, we should add this to the setup.py `install_requires`. But optional dependency seems nicer.

Reply:
+1, let's conditionally import, like:

```python
try:
    import huggingface_hub
except ImportError:
    huggingface_hub = None

...

# Later:
if huggingface_hub is None:
    raise ImportError(
        "To save models to the Hugging Face Hub, "
        "you must install the `huggingface_hub` package."
    )
```

Reply:
Addressed in https://github.com/keras-team/keras/pull/19854/commits/857111272c681bb7ad3d2aa19a82632540d10675. I also removed `huggingface_hub` from `requirements-common.txt`